steward/src/cork.rs

```diff
@@ -84,11 +99,139 @@ impl steward::contract_call_server::ContractCall for CorkHandler {
                 "failed to send cork to sommelier",
             ));
         }
-        info!("submitted cork for {}!", cellar_id);
+        debug!("sent cork!");
+
+        Ok(Response::new(SubmitResponse {}))
+    }
+}
+
+pub struct DirectCorkHandler {
+    pub name: String,
+}
+
+#[async_trait]
+impl steward::contract_call_server::ContractCall for DirectCorkHandler {
+    async fn submit(
+        &self,
+        request: Request<SubmitRequest>,
+    ) -> Result<Response<SubmitResponse>, Status> {
+        debug!("received contract call request");
+        let request = request.get_ref();
+
+        let contract_call_data = match request.call_data.clone() {
+            Some(call) => call,
+            None => {
+                return Err(tonic::Status::invalid_argument(
+                    "Error, can't find call data",
+                ))
+            }
+        };
+
+        let config = APP.config();
+
+        let name = self.name.clone();
+        let wallet = config.load_ethers_wallet(name);
+
+        let eth_host = config.ethereum.rpc.clone();
+        let provider = Provider::<Http>::try_from(eth_host.clone())
+            .unwrap_or_else(|_| panic!("could not get provider from eth_host {}", eth_host));
+        let chain_id = provider
+            .get_chainid()
+            .await
+            .expect("Could not retrieve chain ID");
+
+        let chain_id =
+            downcast_to_u64(chain_id).expect("Chain ID overflowed when downcasting to u64");
+        let client = SignerMiddleware::new(provider, wallet.clone().with_chain_id(chain_id));
+
+        let cellar_address = request.cellar_id.clone();
+
+        if cellars::validate_cellar_id(&cellar_address).is_err() {
+            return Err(tonic::Status::invalid_argument(
+                "Error, can't validate Cellar ID",
+            ));
+        }
+
+        let address: Address = cellar_address
+            .parse()
+            .expect("could not parse cellar address");
+
+        let function_call = match contract_call_data {
+            AaveV2Stablecoin(call) => {
+                let contract = AaveV2StablecoinCellar::new(address, Arc::new(client));
+                match call.function.unwrap() {
+                    Function::EnterPosition(_) => contract_call(contract.enter_position()).await,
+                    Function::SetDepositLimit(params) => {
+                        contract_call(
+                            contract.set_deposit_limit(string_to_u256(params.limit).unwrap()),
+                        )
+                        .await
+                    }
+                    Function::Reinvest(params) => {
+                        contract_call(
+                            contract.reinvest(string_to_u256(params.min_assets_out).unwrap()),
+                        )
+                        .await
+                    }
+                    Function::Rebalance(params) => {
+                        let results: Vec<Result<H160, &String>> = params
```

Comment:
If we're going to be calling this long list of validation functions in two places, perhaps we should this portion into a separate function in the Aave cellar code and then we could call that from here.